test/e2e/preemption/preemption_test.go

```diff
@@ -328,6 +335,223 @@ var _ = ginkgo.Describe("Preemption", func() {
+		annotation = "ann-" + common.RandSeq(10)
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+
+			var err error
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:       "sandbox1",
+				Resources:  configs.Resources{Guaranteed: map[string]string{"memory": fmt.Sprintf("%dM", sleepPodMemLimit2)}},
+				Properties: map[string]string{"preemption.delay": "1s"},
+			}); err != nil {
+				return err
+			}
+
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:       "sandbox2",
+				Resources:  configs.Resources{Guaranteed: map[string]string{"memory": fmt.Sprintf("%dM", sleepPodMemLimit2)}},
+				Properties: map[string]string{"preemption.delay": "1s"},
+			}); err != nil {
+				return err
+			}
+
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:       "sandbox3",
+				Resources:  configs.Resources{Guaranteed: map[string]string{"memory": fmt.Sprintf("%dM", sleepPodMemLimit2)}},
+				Properties: map[string]string{"preemption.delay": "1s"},
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+
+		// Define PriorityClass
+		var preemptAllowPriorityClass = schedulingv1.PriorityClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "allow-preemption",
+				Annotations: map[string]string{constants.AnnotationAllowPreemption: constants.True},
+			},
+		}
+		var preemptNotAllowPriorityClass = schedulingv1.PriorityClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "preemption-not-allow",
+				Annotations: map[string]string{constants.AnnotationAllowPreemption: constants.False},
+			},
+		}
+
+		// Create PriorityClass
+		ginkgo.By(fmt.Sprintf("Creating priority class %s", preemptAllowPriorityClass.Name))
+		_, err := kClient.CreatePriorityClass(&preemptAllowPriorityClass)
+		gomega.Ω(err).ShouldNot(HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Creating priority class %s", preemptNotAllowPriorityClass.Name))
+		_, err = kClient.CreatePriorityClass(&preemptNotAllowPriorityClass)
+		gomega.Ω(err).ShouldNot(HaveOccurred())
+
+		// Define sleepPod
+		sleepPod1Config := k8s.SleepPodConfig{Name: "sleepjob1", NS: dev, Mem: sleepPodMemLimit2, Time: 60, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod2Config := k8s.SleepPodConfig{Name: "sleepjob2", NS: dev, Mem: sleepPodMemLimit2, Time: 60, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod3Config := k8s.SleepPodConfig{Name: "sleepjob3", NS: dev, Mem: sleepPodMemLimit2, Time: 60, Optedout: false, Labels: map[string]string{"queue": "root.sandbox2"}}
+		sleepPod4Config := k8s.SleepPodConfig{Name: "sleepjob4", NS: dev, Mem: sleepPodMemLimit2, Time: 60, Optedout: false, Labels: map[string]string{"queue": "root.sandbox2"}}
```

Comment:
The `Optedout` config didn't take effect. I fix it in https://github.com/apache/yunikorn-k8shim/pull/646. If we want to test `yunikorn.apache.org/allow-preemption` annotation key in PriorityClass, we may need to remove Pod annotations and only keep PriorityClassName, because we check annotations from Pod first and then PriorityClass.

https://github.com/apache/yunikorn-k8shim/blob/ac8df7992c94b9b5c36b28452e2de1b5a050f373/pkg/cache/task.go#L251-L261